tests/llm/nodes/test_langchain_agent_node.py

```diff
@@ -143,3 +155,81 @@ def test_execute_error(mock_chat_completion: tuple[mock.MagicMock, mock.MagicMoc
+
+
+class MetaDataAgentNode(LangChainAgentNode):
+    """
+    Subclass of LangChainAgentNode that allows for the passing of metadata to the agent.
+    """
+
+    def __init__(self, agent_executor: "AgentExecutor", metadata: dict[str, typing.Any]):
+        super().__init__(agent_executor=agent_executor)
+        self._metadata = metadata
+
+    async def execute(self, context: LLMContext) -> LLMContext:
+
+        input_dict = context.get_inputs()
+        results = await self._run_single(metadata=self._metadata, **input_dict)
+
+        context.set_output(results)
+
+        return context
+
+
+class MetadataSaverTool(BaseTool):
+    # The base class defines *args and **kwargs in the signature for _run and _arun requiring the arguments-differ
+    # pylint: disable=arguments-differ
+    name: str = "MetadataSaverTool"
+    description: str = "useful for when you need to know the name of a reptile"
+
+    saved_metadata: dict = {}
+
+    def _run(
+        self,
+        query: str,
+        run_manager: typing.Optional[CallbackManagerForToolRun] = None,
+    ) -> str:
+        assert query is not None  # avoiding unused-argument
+        assert run_manager is not None
+        self.saved_metadata.update(run_manager.metadata)
+        return "frog"
+
+    async def _arun(
+        self,
+        query: str,
+        run_manager: typing.Optional[AsyncCallbackManagerForToolRun] = None,
+    ) -> str:
+        assert query is not None  # avoiding unused-argument
+        assert run_manager is not None
+        self.saved_metadata.update(run_manager.metadata)
+        return "frog"
+
+
+def test_metadata(mock_chat_completion: tuple[mock.MagicMock, mock.MagicMock]):
+    # Tests the execute method of the LangChainAgentNode with a a mocked tools and chat completion
+    (_, mock_async_client) = mock_chat_completion
+    chat_responses = [
+        'I should check Tool1\nAction: MetadataSaverTool\nAction Input: "name a reptile"',
+        'Observation: Answer: Yes!\nI now know the final answer.\nFinal Answer: Yes!'
+    ]
+    mock_responses = [mk_mock_openai_response([response]) for response in chat_responses]
+    mock_async_client.chat.completions.create.side_effect = mock_responses
+
+    llm_chat = ChatOpenAI(model="fake-model", openai_api_key="fake-key")
+
+    metadata_saver_tool = MetadataSaverTool()
+
+    tools = [metadata_saver_tool]
+
+    agent = initialize_agent(tools,
+                             llm_chat,
+                             agent=AgentType.ZERO_SHOT_REACT_DESCRIPTION,
+                             verbose=True,
+                             handle_parsing_errors=True,
+                             early_stopping_method="generate",
+                             return_intermediate_steps=False)
+
+    node = MetaDataAgentNode(agent_executor=agent, metadata={"morpheus": "unittest"})
+
+    assert execute_node(node, input="input1") == "Yes!"
+    assert metadata_saver_tool.saved_metadata == {"morpheus": "unittest"}
```

Comment:
I'd also like to see tests with more nested input since the agent node can flatten out inputs.